shopfloor_reception_shipment_advice/services/reception.py

```diff
@@ -0,0 +1,212 @@
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    # Recover from the header the current shipment ID being worked on.
+    #
+    # If the move being received was choosen from a shipment advice when
+    # redirected back to a move selection, this allows to display the shipment
+    # advice related data.
+
+    @property
+    def _validation_rules(self):
+        return super()._validation_rules + (
+            (self.SHIPMENT_ID_HEADER_RULE, self._is_active_header_shipment),
+        )
+
+    SHIPMENT_ID_HEADER_RULE = (
+        # header name, coerce func, ctx handler, mandatory
+        "HTTP_SERVICE_CTX_SHIPMENT_ID",
+        int,
+        "_work_ctx_get_shipment_id",
+        False,
+    )
+
+    def _work_ctx_get_shipment_id(self, rec_id):
+        return (
+            "current_shipment_advice",
+            self.env["shipment.advice"].browse(rec_id).exists(),
+        )
+
+    def _is_active_header_shipment(self, request, method):
+        return True
+
+    @property
+    def shipment_advice(self):
+        if hasattr(self.work, "current_shipment_advice"):
+            return self.work.current_shipment_advice
+        return self.env["shipment.advice"]
+
+    def _domain_shipment_advice(self, dock, today_only=False):
+        domain = [
+            ("state", "in", ["in_progress", "confirmed"]),
+            ("warehouse_id", "in", self.picking_types.warehouse_id.ids),
+            ("shipment_type", "=", "incoming"),
+        ]
+        if dock:
+            domain.append(("dock_id", "=", dock.id))
+        if today_only:
+            today_start, today_end = self._get_today_start_end_datetime()
+            domain.append(("arrival_date", ">=", today_start))
+            domain.append(("arrival_date", "<=", today_end))
+        return domain
+
+    def _scan_document__by_dock(self, dock, barcode):
+        if not dock:
+            return
+        return self._response_for_manual_selection_shipment(dock)
+
+    def _scan_document__by_shipment(self, shipment, barcode):
+        if not shipment:
+            return
+        if shipment.shipment_type != "incoming":
+            return self._response_for_select_document(
+                message=self.msg_store.shipment_incoming_type_only()
+            )
+        if shipment.state not in ["confirmed", "in_progress"]:
+            return self._response_for_select_document(
+                message=self.msg_store.shipment_not_ready(shipment)
+            )
+        planned_pickings = shipment.planned_picking_ids
+        pickings = planned_pickings.filtered(
+            lambda p: p.picking_type_id.id in self.picking_types.ids
+        )
+        if not pickings:
+            return self._response_for_select_document(
+                message=self.msg_store.shipment_nothing_to_unload(shipment)
+            )
+        self.work.current_shipment_advice = shipment
+        return self._response_for_select_move(picking=pickings)
```

Comment:
semantic of _response_for_select_move expect one picking?

Reply:
Yes. On the select_move screen, the picking is supposed to be selected, and you're supposed to select a move from it.
I suppose there might be multiple planned pickings at this point.